qa/rpc-tests/rpcnamedargs.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python3
+# Copyright (c) 2016 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+from decimal import Decimal
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.authproxy import JSONRPCException
+from test_framework.util import (
+    assert_equal,
+    assert_raises_jsonrpc,
+    assert_is_hex_string,
+    assert_is_hash_string,
+    start_nodes,
+    connect_nodes_bi,
+)
+
+
+class NamedArgumentTest(BitcoinTestFramework):
+    """
+    Test named arguments on RPC calls.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.setup_clean_chain = False
+        self.num_nodes = 1
+
+    def setup_network(self, split=False):
+        self.nodes = start_nodes(self.num_nodes, self.options.tmpdir)
+        self.is_network_split = False
+        self.sync_all()
+
+    def run_test(self):
+        node = self.nodes[0]
+        h = node.help(command='getinfo')
+        assert(h.startswith('getinfo\n'))
+
+        assert_raises_jsonrpc(-8, node.help, random='getinfo')
+
+        h = node.getblockhash(index=0)
+        node.getblock(hash=h)
+
+        assert_equal(node.echo(), [])
+        assert_equal(node.echo(arg0=0,arg9=9), [0] + [None]*8 + [9])
```

Comment:
I added the test: 
assert_equal(node.echo(arg0=0,arg3=3,arg9=9), [0] + [None]*2 + [3] + [None]*5 + [9])
Verified that with the code as is it fails, but with instagibbs fix of hole=0 it succeeds.